static/js/pages/storages/StorageUsedBy.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { FC, Fragment, useState } from "react";
+import { Link } from "react-router-dom";
+import { List, Tabs } from "@canonical/react-components";
+import ImageName from "pages/images/ImageName";
+import { LxdStorage } from "types/storage";
+
+interface UsedByType {
+  name: string;
+  project: string;
+  instance: string;
+}
+
+interface Props {
+  storage: LxdStorage;
+  project: string;
+}
+
+const INSTANCES = "Instances";
+const PROFILES = "Profiles";
+const IMAGES = "Images";
+const SNAPSHOTS = "Snapshots";
+const TABS = [INSTANCES, PROFILES, IMAGES, SNAPSHOTS];
+
+const StorageUsedBy: FC<Props> = ({ storage, project }) => {
+  const [activeTab, setActiveTab] = useState(INSTANCES);
+
+  const filterUsedByType = (type: string) =>
+    storage.used_by
+      ?.filter((path) => {
+        if (type === "instances" && path.includes("/snapshots/")) {
+          return false;
+        }
+
+        if (type === "snapshots") {
+          return path.includes("/snapshots/");
+        }
+
+        return path.startsWith(`/1.0/${type}`);
+      })
+      .map((path) => {
+        const name: string = path.split("/").slice(-1)[0] ?? "";
+        return {
+          name: name.split("?")[0],
+          project: name.includes("?project=")
+            ? name.split("=").slice(-1)[0]
+            : project,
+          instance: type === "snapshots" ? path.split("/")[3] : "",
```

Comment:
I propose adding one or more comments above/inside this chunk with one or more example paths to give some context, so that when one reads it they can figure out what's happening more easily and without recurring to external sources (the API specs).

Reply:
Agree, this was quite hacky, I changed it to be more straight forward. Also extracted this into a util file, so we can reuse it for the other pages.